Python/_warnings.c

```diff
@@ -1254,6 +1654,36 @@ PyErr_WarnExplicit(PyObject *category, const char *text,
     return ret;
 }
 
+int
+PyErr_WarnExplicit_WithFix(PyObject *category, const char *msg_txt, const char *fix_txt,
+                   const char *filename_str, int lineno,
+                   const char *module_str, PyObject *registry)
+{
+    PyObject *message = PyUnicode_FromString(msg_txt);
+    PyObject *fix = PyUnicode_FromString(fix_txt);
+    PyObject *filename = PyUnicode_DecodeFSDefault(filename_str);
+    PyObject *module = NULL;
+    int ret = -1;
+
+    if (message == NULL || filename == NULL || fix == NULL)
+        goto exit;
+    if (module_str != NULL) {
+        module = PyUnicode_FromString(module_str);
+        if (module == NULL)
+            goto exit;
+    }
+
+    ret = PyErr_WarnExplicitWithFixObject(category, message, fix, filename, lineno,
+                                   module, registry);
+
+ exit:
+    Py_XDECREF(message);
```

Comment:
I can't remember if `XDECREF` is safe if a variable is `null` (I guess it is).

Reply:
`PyObject *message = PyUnicode_FromString(msg_txt);` looks to be filling it, no?